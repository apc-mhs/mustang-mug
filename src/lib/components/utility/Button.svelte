<script>
	export let disabled; 
</script>

<button on:click disabled={disabled}>
    <slot></slot>
</button>

<style>
button {
    margin-top: 0.5em;
	background-color: rgb(206, 44, 44);
	border-radius:3px;
	border:1px solid #0b0e07;
	display:inline-block;
	cursor:pointer;
	color:#ffffff;
	font-size:15px;
	padding:9px 23px;
	text-decoration:none;
	text-shadow:0px 1px 0px black;
}
button:hover {
	background-color:rgb(126, 33, 33);
}
button:active {
	position:relative;
	top:1px;
}
</style>
