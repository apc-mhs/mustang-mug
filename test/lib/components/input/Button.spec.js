import Button from '$lib/components/input/Button.svelte';
import { render } from '@testing-library/svelte';

describe('Button', () => {
    it('renders correctly', () => {
        const tree = render(Button).container;
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when disabled', () => {
        const tree = render(Button, {
            disabled: true
        }).container;
        expect(tree).toMatchSnapshot();
    });
});
