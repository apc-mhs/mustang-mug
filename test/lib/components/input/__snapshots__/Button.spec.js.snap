// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.svelte renders correctly 1`] = `
<body>
  <div>
    <button
      class="button svelte-43imro"
    />
  </div>
</body>
`;

exports[`Button.svelte renders correctly when disabled 1`] = `
<body>
  <div>
    <button
      class="button svelte-43imro"
      disabled=""
    />
  </div>
</body>
`;
